### PR TITLE
`hedera-app` Part 2

### PR DESCRIPTION
**Pull request by mhess-swl**
PBJ integration for more of the production files and a lot of tests in `hedera-app`.

Linked to https://github.com/hashgraph/hedera-services/issues/5616 (but does not close it)

### CODE REVIEW

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryWorkflowImplTest.java [hunk 1]
```diff
@@ -464,526 +470,140 @@ void testInactivePlatformFails(@Mock CurrentPlatformStatus localCurrentPlatformS
...
-    //        final var response = parseResponse(responseBuffer);
-    //        assertThat(response.fileGetInfo()).isNotNull();
-    //        final var header = response.fileGetInfoOrThrow().header();
-    //        assertThat(header.nodeTransactionPrecheckCode())
-    //                .isEqualTo(PLATFORM_TRANSACTION_NOT_CREATED);
-    //        assertThat(header.responseType()).isEqualTo(ANSWER_ONLY);
-    //        assertThat(header.cost()).isZero();
-    //        verify(opCounters).countReceived(FileGetInfo);
-    //        verify(opCounters, never()).countAnswered(FileGetInfo);
-    //    }
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/query/QueryWorkflowImplTest.java:937]: These tests were duplicates. There's another test for each one of these with the same name

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/MonoGetTopicInfoUsage.java [hunk 2]
```diff
@@ -75,16 +76,15 @@ public FeeData computeUsage(final Query query, final ReadableTopicStore topicSto
...
                 topic.memo(),
-                (JKey) fromPbjKey(topic.adminKey()).orElse(null),
-                (JKey) fromPbjKey(topic.submitKey()).orElse(null),
+                (JKey) fromPbjKey(topic.adminKeyOrElse(Key.DEFAULT)).orElse(null),
+                (JKey) fromPbjKey(topic.submitKeyOrElse(Key.DEFAULT)).orElse(null),
                 topic.autoRenewPeriod(),
                 new EntityId(0, 0, topic.autoRenewAccountNumber()),
                 new RichInstant(topic.expiry(), 0));
-        monoTopic.setRunningHash(unwrapPbj(topic.runningHash()));
+        monoTopic.setRunningHash(PbjConverter.asBytes(topic.runningHash()));
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/MonoGetTopicInfoUsage.java:87]: We should probably delete `unwrapPbj` if its not being used anywhere

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java [hunk 3]
```diff
@@ -92,23 +93,25 @@ public void submit(@NonNull final TransactionBody txBody, @NonNull final byte[]
 
         // Unchecked submits are a mechanism to inject transaction to the system, that bypass all
         // pre-checks.This is used in tests to check the reaction to illegal input.
-        final var optUncheckedSubmit = txBody.uncheckedSubmit();
-        if (optUncheckedSubmit.isPresent()) {
+        final var uncheckedSubmit = txBody.uncheckedSubmitOrElse(UncheckedSubmitBody.DEFAULT);
+        if (!uncheckedSubmit.equals(UncheckedSubmitBody.DEFAULT)) {
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java:97]: Shouldn't we just do `op.hasUncheckedSubmit` here ?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java [hunk 4]
```diff
@@ -92,23 +93,25 @@ public void submit(@NonNull final TransactionBody txBody, @NonNull final byte[]
...
+            LOG.warn("Unchecked submit is not supported in this version of Hedera");
             if (nodeLocalProperties.activeProfile() == Profile.PROD) {
                 // we do not allow unchecked submits in PROD
                 throw new PreCheckException(PLATFORM_TRANSACTION_NOT_CREATED);
             }
-            final var uncheckedSubmit = optUncheckedSubmit.get();
             final var uncheckedTxBytes = uncheckedSubmit.transactionBytes();
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java:103]: Then here `final var uncheckedSubmit = txBody.uncheckedSubmitOrThrow();`

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java [hunk 5]
```diff
@@ -164,19 +165,19 @@ private TransactionMetadata preHandle(
...
             TransactionMetadata innerMetadata = null;
             if (innerContext != null) {
-                final var innerTxBodyBytes = innerContext.getTxn().toByteArray();
+                // VERIFY: the txBytes used for inner transactions is the same as the outer transaction
                 final var innerPayerSignature =
-                        verifyPayerSignature(state, innerContext, innerTxBodyBytes, signatureMap);
-                final var innerOtherSignatures =
-                        verifyOtherSignatures(state, innerContext, innerTxBodyBytes, signatureMap);
+                        verifyPayerSignature(state, innerContext, Bytes.wrap(txBytes), signatureMap);
+                final var innerOtherSignatures = verifyOtherSignatures(state, innerContext, txBytes, signatureMap);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java:180]: Seems inconsistent that we are using `Bytes.wrap` for one method and not for other. We should probably change to `Bytes` on line 153

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/MonoGetTopicInfoUsage.java (hunk 2 above)
Comment (mhess-swl) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/MonoGetTopicInfoUsage.java:87]: Agreed, removed

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java (hunk 5 above)
Comment (mhess-swl) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java]: done!

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java (hunk 4 above)
Comment (mhess-swl) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java:102]: done!

**Review by rbair23 — APPROVED** [review]

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/MerkleTestBase.java [hunk 6]
```diff
@@ -300,31 +300,28 @@ protected <T extends MerkleNode> T parseTree(@NonNull final byte[] state, @NonNu
...
-            output.writeBytes(bytes);
+        public String parseStrict(@NonNull ReadableSequentialData input) {
+            return parse(input);
         }
 
         @Override
-        public int measure(@NonNull DataInput input) throws IOException {
-            return input.readInt();
-        }
+        public void write(@NonNull String s, @NonNull WritableSequentialData output) {}
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/MerkleTestBase.java:320]: Is this intentional to leave the method body blank ? Seems its overridden twice.  Did the IDE not show error ?

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Have minor comments

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/serdes/MonoSchedulingStateAdapterCodec.java [hunk 7]
```diff
@@ -59,12 +52,14 @@ public int measure(@NonNull ReadableSequentialData input) {
...
+        return 0;
     }
 
     @Override
-    public int measureRecord(MerkleScheduledTransactionsState merkleScheduledTransactionsState) {
-        return 0;
+    public boolean fastEquals(
+            @NonNull MerkleScheduledTransactionsState merkleScheduledTransactionsState,
+            @NonNull ReadableSequentialData readableSequentialData) {
+        throw new UnsupportedOperationException("Not implemented yet");
```
[the remark this answers is not in the document]
Comment (mhess-swl) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/serdes/MonoSchedulingStateAdapterCodec.java:63]: I'm not worrying about the warnings until after the tests are passing. This is not the last PR for `hedera-app`

[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/serdes/MonoSchedulingStateAdapterCodec.java:63]: I see. Makes sense

**Pull request merged**